crates/fluvio-dataplane-protocol/src/batch.rs

```diff
@@ -356,6 +357,14 @@ impl BatchHeader {
         let compression_bits = compression as i16 & COMPRESSION_CODEC_MASK;
         self.attributes = (self.attributes & !COMPRESSION_CODEC_MASK) | compression_bits;
     }
+
+    pub fn set_first_timestamp(&mut self, timestamp: i64) {
```

Comment:
this should probably not make it public since it's internal

Reply:
Not sure how to set timestamp_delta without making this public. That is being set by accumulator, and it needs to know the batch creation timestamp

Reply:
that can be a time of inserting the record.  first time stamp if the batch is empty otherwise it will be the max time stamp?

Reply:
But all these things are done by the `BatchingProducer` struct that is in the fluvio crate.  It needs to set the timestamp_delta before inserting the record because timestamp_delta is varint and the size of the record (which is needed before inserting) could change because of that